src/settings/config.rs

```diff
@@ -0,0 +1,89 @@
+// Config file handling
+
+use std::env;
+
+use serde::Deserialize;
+
+use crate::{dimensions::Dimensions, frame::Frame};
+
+use std::path::{Path, PathBuf};
+
+const CONFIG_FILE: &str = "config.toml";
+
+#[cfg(unix)]
+fn neovide_config_dir() -> PathBuf {
+    let xdg_dirs = xdg::BaseDirectories::with_prefix("neovide").unwrap();
+    xdg_dirs.get_config_home()
+}
+
+#[cfg(windows)]
+fn neovide_config_dir() -> PathBuf {
+    let mut path = dirs::config_dir().unwrap();
+    path.push("neovide");
+    path
+}
+
+pub fn config_path() -> PathBuf {
+    let mut config_path = neovide_config_dir();
+    config_path.push(CONFIG_FILE);
+    config_path
+}
+
+#[derive(Debug, Deserialize, Default)]
+pub struct Config {
+    pub multigrid: Option<bool>,
+    pub maximized: Option<bool>,
+    pub vsync: Option<bool>,
+    pub srgb: Option<bool>,
+    pub no_idle: Option<bool>,
+    pub neovim_bin: Option<PathBuf>,
+    pub frame: Option<Frame>,
+    pub geometry: Option<Dimensions>,
+}
+
+impl Config {
+    pub fn write_to_env(&self) {
+        if let Some(multigrid) = self.multigrid {
+            env::set_var("NEOVIDE_MULTIGRID", multigrid.to_string());
+        }
+        if let Some(maximized) = self.maximized {
+            env::set_var("NEOVIDE_MAXIMIZED", maximized.to_string());
+        }
+        if let Some(vsync) = self.vsync {
+            env::set_var("NEOVIDE_VSYNC", vsync.to_string());
+        }
+        if let Some(srgb) = self.srgb {
+            env::set_var("NEOVIDE_SRGB", srgb.to_string());
+        }
+        if let Some(no_idle) = self.no_idle {
+            env::set_var("NEOVIDE_NO_IDLE", no_idle.to_string());
```

Comment:
we're changed variable name to `NEOVIDE_IDLE` recently, so, behaviour is reversed. We should use `idle: false`  here too.

Reply:
Actually I did not change `NEOVIDE_IDLE` only `NEOVIDE_SRGB` and `NEOVIDE_VSYNC`. But it would probably make sense to change it the same way.

Reply:
Actually, you did change `NEOVIDE_IDLE` and it's broken now. I created a PR to fix that:  #1889. I'll update the code here as well.